web-api/benches/stateless_personalization.rs

```diff
@@ -15,35 +15,35 @@
 use std::hint::black_box;
 
 use chrono::Utc;
-use criterion::{criterion_group, criterion_main, Criterion};
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
 use itertools::Itertools;
 use xayn_ai_bert::Embedding1;
 use xayn_ai_coi::CoiConfig;
 use xayn_web_api::bench_derive_interests;
 
-fn derive_interests_with_identical_embeddings(embedding_size: usize, interest_size: usize) {
-    let system = CoiConfig::default().build();
-    let timestamp = Utc::now();
-    let embedding = Embedding1::from(vec![1.0; embedding_size])
-        .normalize()
-        .unwrap();
-    let history = vec![(timestamp, embedding); interest_size];
-    bench_derive_interests(black_box(&system), black_box(history));
-}
-
 macro_rules! bench_identical {
     ($($function: ident, $embedding_size: expr, $interest_size: expr);+ $(;)?) => {$(
         fn $function(c: &mut Criterion) {
+            let system = CoiConfig::default().build();
+            let timestamp = Utc::now();
+            let embedding = Embedding1::from(vec![1.0; $embedding_size])
+                .normalize()
+                .unwrap();
+            let history = vec![(timestamp, embedding); $interest_size];
+
             let name = format!(
                 "derive {} interests with identical embeddings ({})",
                 $interest_size,
                 $embedding_size,
             );
-            c.bench_function(&name, |b| {
-                b.iter(|| {
-                    derive_interests_with_identical_embeddings($embedding_size, $interest_size)
-                })
-            });
+            c.bench_function(
+                &name,
+                |b| b.iter_batched(
+                    || history.clone(),
+                    |history| bench_derive_interests(black_box(&system), black_box(history)),
```

Comment:
art

there is no reason to black_box the history , or any "batch" data passed with iter_batched

```suggestion
                    |history| bench_derive_interests(black_box(&system), history),
```